### PR TITLE
Added Scheduler to SDXL demo

### PR DESCRIPTION
**Pull request by ipotkonjak-tt**
### What's changed
Added device implementation for scheduler to SDXL demo. Additionally, changed the scheduler test to use channel last format, since neither reference or ttnn implementation assume a specific dimension order.

### Checklist
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/runs/15070033927) CI passes
- [x] [(Single-card) Frequent model and ttnn tests](https://github.com/tenstorrent/tt-metal/actions/runs/15070038212) CI passes

### CODE REVIEW

**Review by pavlepopovic — APPROVED** [review]

File: models/experimental/stable_diffusion_xl_base/demo.py [hunk 1]
```diff
@@ -314,68 +342,98 @@ def run_demo_inference(
...
 
-    if vae_on_device:
-        # Workaround for #22017
-        ttnn_device.disable_and_clear_program_cache()
+        latents = tt_scheduler.step(
+            noise_pred, tt_scheduler.timesteps[i], latents, **extra_step_kwargs, return_dict=False
+        )[0]
+
+        ttnn.deallocate(noise_pred)
+        latents = ttnn.move(latents)
```
[the remark this answers is not in the document]
Comment (ipotkonjak-tt) [reply models/experimental/stable_diffusion_xl_base/demo.py:410]: We hit OOM because of fragmentation if we don't move

File: models/experimental/stable_diffusion_xl_base/demo.py [hunk 2]
```diff
@@ -74,33 +75,37 @@ def retrieve_timesteps(
...
-        dtype=ttnn.bfloat16,
-        device=ttnn_device,
-        layout=ttnn.TILE_LAYOUT,
-        memory_config=ttnn.DRAM_MEMORY_CONFIG,
-    )
-
-    ttnn_latent_model_input = ttnn.permute(ttnn_latent_model_input, (0, 2, 3, 1))
-    ttnn_latent_model_input = ttnn.reshape(ttnn_latent_model_input, (1, 1, B * H * W, C))
+    B, C, H, W = input_shape
+    if isinstance(input_tensor, torch.Tensor):
```
[the remark this answers is not in the document]
Comment (ipotkonjak-tt) [reply models/experimental/stable_diffusion_xl_base/demo.py:90]: No, this is leftover code from comparing with reference scheduler. Will remove it

File: models/experimental/stable_diffusion_xl_base/demo.py [hunk 3]
```diff
@@ -170,7 +195,10 @@ def run_demo_inference(
     )
 
     # Prepare timesteps
-    timesteps, num_inference_steps = retrieve_timesteps(pipeline.scheduler, num_inference_steps, cpu_device, None, None)
+    _, _ = retrieve_timesteps(
```
[the remark this answers is not in the document]
Comment (ipotkonjak-tt) [reply models/experimental/stable_diffusion_xl_base/demo.py:198]: we can, but for now we keep schedulers `set_timesteps` function on cpu since it is not a part of the unet loop. If we decide to move this function to device then we can just use device that is passed `TtAutoencoderKL` in init.

File: models/experimental/stable_diffusion_xl_base/demo.py [hunk 4]
```diff
@@ -170,7 +195,10 @@ def run_demo_inference(
...
+        pipeline.scheduler, num_inference_steps, cpu_device, None, None
+    )  # Bad output if this is removed, implicit dependency exists
+    timesteps, num_inference_steps = retrieve_timesteps(tt_scheduler, num_inference_steps, cpu_device, None, None)
```
[the remark this answers is not in the document]
Comment (ipotkonjak-tt) [reply models/experimental/stable_diffusion_xl_base/demo.py:201]: The order is irrelevant, but for the image to be generated properly we need the cpu `retrieve_timesteps()` call.

This is caused by `pipeline` using scheduler attributes set in this function. For example in `prepare_latents()` we have `latents = latents * self.scheduler.init_noise_sigma`. This is still the cpu scheduler and `init_noise_sigma` is actually a function that needs `scheduler.sigmas` to be set

[the remark this answers is not in the document]
Comment (ipotkonjak-tt) [reply models/experimental/stable_diffusion_xl_base/demo.py:193]: Resolved this by swapping the cpu scheduler in pipeline with our implementation
